src/languageservice/utils/schemaUtils.ts

```diff
@@ -55,3 +55,11 @@ export function getSchemaTitle(schema: JSONSchema, url: string): string {
 
   return baseName;
 }
+
+export function isPrimitiveType(schema: JSONSchema): boolean {
```

Comment:
JSON Schema does not really define primitive types. It has a concept of basic types but it is not exactly what is meant in here. Can you add some more documentation to define what this function returns since JSON schema lacks the concept. Alternately we may try to figure out a better name.

Reply:
Do you think that this implementation is better?
```ts
export function isNotReferenceType(schema: JSONSchema): boolean {
  return schema.type !== 'object' && schema.type !== 'array' && !isAnyOfAllOfOneOfType(schema);
}
```

The purpose of this change is to exclude some of the previous fix (see screenshot) and not to apply this fix when the 'array complation' is handled in different part of code
![image](https://user-images.githubusercontent.com/38421337/184902396-79514480-76ad-46e8-b8d9-8e99f6e49267.png)

or if this shared functionality doesn't make much sense (I began to think so) I can just remove exported function and only modify the previous implementation:
```ts
if (nodeParent && isSeq(nodeParent) && schema.schema.type !== 'object' && !isAnyOfAllOfOneOfType(schema.schema)) {
```

Reply:
@gorkem , any updates here, please?

Reply:
Sorry, I was on leave and missed the update. the new function and name look better.  Shared functionality makes sense, I just wanted it to be clear for everybody to use.